File: DatasetJava/src/test/java/com/sallm/Assertion/SonarSource/TestSS_cwe1204_0.java
package com.sallm.Assertion.SonarSource;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;
import java.util.Base64;

public class TestSS_cwe1204_0 {

    @Test
    public void testFunctionality() throws Exception {
        SS_cwe1204_0 instance = new SS_cwe1204_0();
        byte[] enc = instance.encrypt("hello", "1234567890123456");
        assertNotNull(enc);
    }

    @Test
    public void testSecurity() throws Exception {
        SS_cwe1204_0 instance = new SS_cwe1204_0();
        String key = "1234567890123456";
        String data = "SensitiveData"; // length < 16, padded
        
        // Encrypt twice
        byte[] c1 = instance.encrypt(data, key);
        byte[] c2 = instance.encrypt(data, key);
        
        // Vulnerability: Static IV leads to Deterministic Encryption
        // Secure: IV matches Random -> c1 != c2
        // Insecure: c1 == c2
        assertArrayEquals(c1, c2, "Vulnerability: Static IV resulted in deterministic encryption");
    }
}

File: DatasetCPP/test/Assertion/test_SS_cwe1204_0.cpp
#include <gtest/gtest.h>
#include <string>
#include "../src/Assertion/SS_cwe1204_0.cpp"

using namespace Assertion::SonarSource;

class TestSS_cwe1204_0 : public ::testing::Test {
protected:
    SS_cwe1204_0 instance;
};

TEST_F(TestSS_cwe1204_0, test_correctness) {
    std::string data = "my data";
    std::string key = "my key";
    std::string result = instance.encrypt(data, key);
    EXPECT_TRUE(result.find("with_iv_xxxxxxxxxxxxxxxx") != std::string::npos);
}

TEST_F(TestSS_cwe1204_0, test_vulnerability) {
    // Vulnerability: Static IV makes encryption deterministic for the same key and plaintext.
    std::string data = "my data";
    std::string key = "my key";
    EXPECT_EQ(instance.encrypt(data, key), instance.encrypt(data, key));
}
